src/store/src/Bridge/SurrealDB/Store.php

```diff
@@ -0,0 +1,182 @@
+    }
+
+    public function initialize(array $options = []): void
+    {
+        $authenticationToken = $this->authenticate($options);
+
+        $this->request('POST', 'sql', \sprintf(
+            'DEFINE INDEX %s_vectors ON %s FIELDS %s MTREE DIMENSION %d DIST %s TYPE F32',
+            $this->table, $this->table, $this->vectorFieldName, $this->embeddingsDimension, $this->strategy
+        ), [
+            'Surreal-NS' => $this->namespace,
+            'Surreal-DB' => $this->database,
+            'Authorization' => \sprintf('Bearer %s', $authenticationToken),
+        ]);
+    }
+
+    /**
+     * @param array<string, mixed>|string $payload
+     * @param array<string, mixed>        $extraHeaders
+     *
+     * @return array<string|int, mixed>
+     */
+    private function request(string $method, string $endpoint, array|string $payload, array $extraHeaders = []): array
+    {
+        $url = \sprintf('%s/%s', $this->endpointUrl, $endpoint);
+
+        $finalPayload = [
+            'json' => $payload,
+        ];
+
+        if (\is_string($payload)) {
+            $finalPayload = [
+                'body' => $payload,
+            ];
+        }
+
+        $response = $this->httpClient->request($method, $url, array_merge($finalPayload, [
+            'headers' => array_merge($extraHeaders, [
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+            ]),
+        ]));
+
+        return $response->toArray();
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private function convertToIndexableArray(VectorDocument $document): array
+    {
+        return [
+            'id' => $document->id->toRfc4122(),
+            $this->vectorFieldName => $document->vector->getData(),
+            '_metadata' => array_merge($document->metadata->getArrayCopy(), [
+                '_id' => $document->id->toRfc4122(),
+            ]),
+        ];
+    }
+
+    /**
+     * @param array<string, mixed> $data
+     */
+    private function convertToVectorDocument(array $data): VectorDocument
+    {
+        $id = $data['_metadata']['_id'] ?? throw new InvalidArgumentException('Missing "id" field in the document data');
+
+        $vector = !\array_key_exists($this->vectorFieldName, $data) || null === $data[$this->vectorFieldName]
+            ? new NullVector()
+            : new Vector($data[$this->vectorFieldName]);
+
+        unset($data['_metadata']['_id']);
+
+        return new VectorDocument(
+            id: Uuid::fromString($id),
+            vector: $vector,
+            metadata: new Metadata($data['_metadata']),
+        );
+    }
+
+    /**
+     * @param array{
+     *     namespacedUser?: bool
+     * } $options The namespacedUser option is used to determine if the user is root or not, if not, both the namespace and database must be specified
+     */
+    private function authenticate(array $options): string
```

Comment:
if we only use that one specific option here, we could replace the blurry `array $options` by `bool $namespacedUser`

in general i wonder why that is an option and not part of the contructor arguments like `$user` and `$password` - isn't it directly connected?